### PR TITLE
Bugfix/issue 62

### LINKED ISSUE #62: [BUG] @glue42/core - methodDefinition's getServers() listed as property, not method
**Describe the bug**
https://docs.glue42.com/reference/glue/latest/interop/index.html#!MethodDefinition
getServers appears as if it's a property. It actually got me confused as to whether this is indeed a property keeping the server's Instance object.

"getServers?: () => Instance[];" should be "getServers(): Instance[];" in the glue.d.ts file

### PR DESCRIPTION
**Pull request by kirilpopov**
Closes #62

### CODE REVIEW

File: packages/core/src/interop/client/client.ts [hunk 1]
```diff
@@ -311,8 +316,14 @@ export default class Client {
...
+                        ...methodDefinition,
+                        getServers: () => [],
+                        supportsStreaming: false,
+                        objectTypes: methodDefinition.objectTypes ?? [],
+                    };
                     const errorObj: InvocationResult = {
-                        method: methodDefinition,
+                        method,
                         called_with: argumentObj,
                         message: "Can not find a method matching " + JSON.stringify(methodFilter) + " with server filter " + JSON.stringify(target) + ". Is the object a valid instance ?",
```
Comment (gdavidkov) [inline packages/core/src/interop/client/client.ts:328]: Can you revise the error message?

File: packages/core/src/interop/client/client.ts [hunk 2]
```diff
@@ -158,24 +159,28 @@ export default class Client {
...
      */
-    public methods(methodFilter: Glue42Core.AGM.MethodDefinition): Glue42Core.AGM.MethodDefinition[] {
-        // Must not be mutated
-        const filterCopy = { ...methodFilter };
+    public methods(methodFilter: Glue42Core.AGM.MethodDefinition | string): Glue42Core.AGM.Method[] {
+        if (typeof methodFilter === "string") {
+            methodFilter = { name: methodFilter };
+        } else {
+            // Must not be mutated
+            methodFilter = { ...methodFilter };
```
Comment (gdavidkov) [inline packages/core/src/interop/client/client.ts:167]: Do we need a deep clone? I see that we have property from type array - objectTypes?

File: packages/core/src/interop/client/client.ts (hunk 1 above)
Comment (kirilpopov) [reply packages/core/src/interop/client/client.ts:328]: done

File: packages/core/src/interop/client/client.ts (hunk 2 above)
Comment (kirilpopov) [reply packages/core/src/interop/client/client.ts:167]: this is an old code (just added an extra if), we can address in separate issue

Comment (gdavidkov) [reply packages/core/src/interop/client/client.ts:167]: I agree

**Pull request merged**